build_tools/python/e2e_test_framework/definitions/iree_definitions.py

```diff
@@ -78,11 +95,23 @@ class ModuleExecutionConfig(object):
   extra_flags: List[str] = dataclasses.field(default_factory=list)
 
 
-class MLIRDialectType(Enum):
+@dataclass(frozen=True)
+class _MLIRDialectPair(object):
+  """MLIR dialect with its unique artificial id."""
+  # Unique artificial id.
+  id: int
+  # Name of an IREE supported input type (--iree-input-type).
+  dialect_name: str
+
+
+# Please use and update the next id when adding a new dialect, so we won't reuse
+# the old deprecated id.
+# Next ID: 4
+class MLIRDialectType(_MLIRDialectPair, Enum):
   """Imported MLIR dialect type."""
-  LINALG = "linalg"
-  TOSA = "tosa"
-  MHLO = "mhlo"
+  LINALG = (1, "linalg")
+  TOSA = (2, "tosa")
+  MHLO = (3, "mhlo")
```

Comment:
Oh, you're right. For some reason I had the impression that `linalg` is an option of `--iree-input-type`. Renamed to `none` (no input transform needed).

The `ModelSourceType.EXPORTED_LINALG_MLIR` below does make sense to me since it means the source model file is a MLIR file and written only in Linalg (with other common dialects like arith, tensor). Let me know WDYT